### PR TITLE
Feature/fbfw-54-Game-Pieces

### PR DESCRIPTION
**Pull request by jchoi166**
### Changes
This Diff creates a WDIcon for both Army and Fleet game pieces using generated SVGR components. 
The icons take "country" and "iconState" prop that determines the country color and state (selected, hold, dislodged, etc) of the icon. 
This diff also adds all of the country colors and hues to the Palatte configuration in the webdip theme. 

### Jira Link w/ Figma
https://codazen.atlassian.net/jira/software/c/projects/TMDS/boards/262?modal=detail&selectedIssue=FBFW-54

### Notes
It would be possible to combine the army and fleet icons into a single "game piece" icon that also takes some kind of "military" prop that would determine if it was an army or fleet piece. However, while doing so, I noticed that it might make the component a little too verbose with lots of nested ternary/if statements so I opted to separate them out into two different components. It looked cleaner to me that way.

### CODE REVIEW

File: beta-src/src/App.tsx [hunk 1]
```diff
@@ -1,13 +1,17 @@
 import * as React from "react";
 import "./assets/css/App.css";
 import Box from "@mui/material/Box";
+import { ThemeProvider } from "@mui/material";
+import webDiplomacyTheme from "./webDiplomacyTheme";
```
Comment (nebevets) [inline beta-src/src/App.tsx:5]: the theme provider wraps app already in index. we should not be placing it here again. pls revert this file.

**Review by nebevets — CHANGES_REQUESTED** [review]
remove theme stuff from App.tsx. if this breaks your code, there is something else wrong, but it should work because theme is already in index

File: beta-src/src/webDiplomacyTheme.ts [hunk 2]
```diff
@@ -13,6 +13,21 @@ declare module "@mui/material/styles" {
...
+
+  interface PaletteOptions {
+    France: CountryColor;
+    Austria: CountryColor;
+    England: CountryColor;
+    Germany: CountryColor;
+    Russia: CountryColor;
+    Italy: CountryColor;
+    Turkey: CountryColor;
+  }
```
Comment (bryancodazen) [inline beta-src/src/webDiplomacyTheme.ts:29]: create an Enum for the countries and put them in src/enums, then use the enum here. Give PaletteOptions a type forcing it use the enum as a keys. Types should be added to src/types and then imported.

File: beta-src/src/webDiplomacyTheme.ts [hunk 3]
```diff
@@ -13,6 +13,21 @@ declare module "@mui/material/styles" {
     desktop: true;
   }
 
+  interface CountryColor {
+    main: string;
+    light: string;
+  }
```
Comment (bryancodazen) [inline beta-src/src/webDiplomacyTheme.ts:19]: create a type for CountryColor

File: beta-src/src/webDiplomacyTheme.ts [hunk 4]
```diff
@@ -113,6 +128,34 @@ const webDiplomacyTheme = createTheme({
...
+      light: "#BFB3EA",
+    },
+    Italy: {
+      main: "#47D2A0",
+      light: "#C2F0DF",
+    },
+    Turkey: {
+      main: "#F3C400",
+      light: "#FBEBAA",
+    },
```
Comment (bryancodazen) [inline beta-src/src/webDiplomacyTheme.ts:158]: use the country enum and create a type for this and then add it to the theme palette after the declaration.

File: beta-src/src/types.d.ts [hunk 5]
```diff
@@ -1,3 +1,21 @@
...
+    | "Russia"
+    | "Italy"
+    | "Turkey";
+  iconState?:
+    | "none"
+    | "selected"
+    | "hold"
+    | "disbanded"
+    | "dislodged"
+    | "build";
```
Comment (bryancodazen) [inline beta-src/src/types.d.ts:20]: Do we need an enum for this?

**Review by bryancodazen — CHANGES_REQUESTED** [review]

File: beta-src/src/App.tsx (hunk 1 above)
Comment (jchoi166) [reply beta-src/src/App.tsx:5]: Whoops! I totally missed that lol. File has been reverted!

File: beta-src/src/webDiplomacyTheme.ts [hunk 6]
```diff
@@ -50,6 +51,17 @@ const disabledTextSecondary = "#bababa";
 const defaultLineHeight = 1.2;
 const normalFontWeight = 400;
 
+/// Interface and type for Country Color
```
Comment (nebevets) [inline beta-src/src/webDiplomacyTheme.ts:54]: remove

**Review by nebevets — APPROVED** [review]
LGTM

**Comment by bryancodazen** [comment]
Hey Jin, please merge in the latest staging

**Pull request merged**